pkg/kubelet/images/image_manager.go

```diff
@@ -224,3 +312,68 @@ func applyDefaultImageTag(image string) (string, error) {
 	}
 	return image, nil
 }
+
+// isEnsuredBySecret - returns two bools and error
+// 1. pulledBySecret: true if it is in ensured secret pulled image list,
+// the list will clean the image after PullImageSecretRecheckDuration
+// 2. ensuredBySecret: true if the secret for an auth used to pull an
+// image has already been authenticated through a successful pull request
+// and the same auth exists for this podSandbox/image.
+func (m *imageManager) isEnsuredBySecret(imageRef string, image kubecontainer.ImageSpec, pullSecrets []v1.Secret) (pulledBySecret, ensuredBySecret bool, err error) {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+	if imageRef == "" {
+		return
```

Comment:
As mentioned by @rphillips earlier, should ideally return the error or at least log it.

Reply:
Fixed.